src/App.vue

```diff
@@ -345,82 +340,99 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
 
     try {
       this.setCurrentDate(this.dateToUsableString(new Date()))
-
-      // ensure we have a NR number
-      if (!this.nrNumber) {
+      const tempId = this.$route.query?.id
+      // ensure we have a Temporary Registration number
+      if (!tempId) {
         this.nameRequestInvalidType = NameRequestStates.NOT_FOUND
         this.nameRequestInvalidErrorDialog = true
         return // go to finally()
       }
 
-      // ensure user is authorized to use this NR
+      this.setTempId(tempId)
+      // ensure user is authorized to use this IA
       await this.checkAuth().catch(error => {
         console.log('Auth error =', error) // eslint-disable-line no-console
         this.accountAuthorizationDialog = true
         throw error // go to catch()
       })
 
+      try {
+        // Retrieve draft filing if it exists for the IA Number specified
+        let draftFiling = await this.fetchDraft()
+        let emptyFiling = this.buildFiling()
+        if (draftFiling?.incorporationApplication?.nameRequest) {
+          await this.processNameRequest(draftFiling)
+        }
+
+        draftFiling = { ...emptyFiling.filing, ...draftFiling }
+
+        // Parse the draft and update Incorporation data into the store if it exists
+        if (draftFiling) {
+          this.parseDraft(draftFiling)
+        }
+
+        // Initialize Fee Summary
+        this.initEntityFees()
+
+        // Set the resources
+        this.setCertifyStatementResource(CertifyStatementResource.find(x => x.entityType === this.entityType))
+      } catch (error) {
+        console.log('Fetch error =', error) // eslint-disable-line no-console
+        this.fetchErrorDialog = true
+        throw error // go to catch()
+      }
+    } catch (e) {
+      // errors should be handled above
+      // just fall through to finally()
+    } finally {
+      this.haveData = true
+      // wait for things to stabilize, then reset flag
+      Vue.nextTick(() => this.setHaveChanges(false))
+    }
+  }
+
+  /** Set up any required fields/ validation if a Name request is present */
+  private async processNameRequest (filing: any): Promise<void> {
+    try {
+      let nameRequest = filing.incorporationApplication.nameRequest
+
+      // ensure we have an NR number
+      if (!nameRequest.nrNumber) {
+        this.nameRequestInvalidType = NameRequestStates.NOT_FOUND
+        this.nameRequestInvalidErrorDialog = true
+        return // go to finally()
+      }
       // fetch NR data
-      const nrResponse = await this.fetchNameRequest(this.nrNumber).catch(error => {
+      const nrResponse = await this.fetchNameRequest(nameRequest.nrNumber).catch(error => {
         console.log('NR error =', error) // eslint-disable-line no-console
-        this.nameRequestInvalidErrorDialog = true
-        throw error // go to catch()
+        // this.nameRequestInvalidErrorDialog = true
+        // throw error // go to catch()
```

Comment:
Are these comments meant to be left in for future?

Reply:
good catch...weird